modules/unsupported/polylabel/src/main/java/org/geotools/polylabel/Cell.java

```diff
@@ -0,0 +1,109 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *    
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *    
+ */
+package org.geotools.polylabel;
+
+import org.geotools.geometry.jts.GeometryBuilder;
+
+import com.vividsolutions.jts.geom.MultiPolygon;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.operation.distance.DistanceOp;
+
+public class Cell implements Comparable<Cell> {
+  static final private GeometryBuilder gb = new GeometryBuilder();
+  private static final double SQRT2 = 1.4142135623730951;
+  private double x;
+  private double y;
+  private double h;
+  private double d;
+  private double max;
+
+  Cell(double x, double y, double h, MultiPolygon polygon) {
```

Comment:
Comment explaining the class would be nice.  Even if it's only meant for implementation, it will help with maintaining.